src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -298,7 +371,20 @@ public static String getTypeBody() {
    * @return returns the JSON segment to construct a container for each of the tags.
    */
   public static String getContainerSettingJson(int startTagIndex, int numToProccess) {
-    String typeID = "HMS-type-" + getFlexyName();
+    String typeID;
+    if (((TagInfo) TagInfoManager.getTagInfoList().get(i)).getType() == TagType.FLOAT) {
+      typeID = "HMS-number-type-" + getFlexyName();
+      Logger.LOG_INFO("setting container to number " + tagName);
+    } else if (((TagInfo) TagInfoManager.getTagInfoList().get(i)).getType() == TagType.INTEGER) {
+      typeID = "HMS-number-type-" + getFlexyName();
+    } else if (((TagInfo) TagInfoManager.getTagInfoList().get(i)).getType() == TagType.BOOLEAN) {
+      typeID = "HMS-boolean-type-" + getFlexyName();
+    } else if (((TagInfo) TagInfoManager.getTagInfoList().get(i)).getType() == TagType.DWORD) {
```

Comment:
same comment about storing ((TagInfo) TagInfoManager.getTagInfoList().get(i)).getType() in a variable